role/api.py

```diff
@@ -0,0 +1,50 @@
+"""
+Functions for handling roles
+"""
+
+from __future__ import unicode_literals
+
+from django.contrib.auth.models import Group
+from guardian.shortcuts import assign_perm
+
+from role.permissions import RepoPermission, GroupTypes
+
+
+def roles_init_new_repo(repo):
+    """
+    Create new groups for the repo
+    """
+    administrator_group = Group.objects.get_or_create(
+        name=GroupTypes.repo_administrator.format(repo.slug)
+    )[0]
+    curator_group = Group.objects.get_or_create(
+        name=GroupTypes.repo_curator.format(repo.slug)
+    )[0]
+    author_group = Group.objects.get_or_create(
```

Comment:
Maybe we can do something like author_group, _ = Group.obj... instead? Tuple unpacking looks cleaner here, no strong opinion on my part though